metricflow/sql/optimizer/tag_required_column_aliases.py

```diff
@@ -86,17 +87,32 @@ def _search_for_expressions(
 
     @override
     def visit_cte_node(self, node: SqlCteNode) -> None:
-        raise NotImplementedError
+        select_statement = node.select_statement
+        # Copy the tagged aliases from the CTE to the SELECT since when visiting a SELECT, the CTE node (not the SELECT
```

Comment:
Why is it necessary to tag both the CTE node and CTE's select node with the same aliases?

Reply:
Before visiting a node, it's assumed that we have already recorded the column aliases required at the node. So for a SELECT statement, the hierarchy looks like:

```
SELECT node -> CTE node -> SELECT node of the CTE
```

If we don't do this copy, this is what happens:

Let's say we start the process with having recorded that `col_0` is required in `SELECT node`. 

* We visit the `SELECT node` and record that `col_0` is required in  `CTE node`.
* Then we visit the CTE node. At the CTE node, we just go visit the `SELECT node of the CTE`
* At the `SELECT node of the CTE`, then there's no information about which columns were required.

Reply:
This was resolved as a part of changes related to another comment.